RecoParticleFlow/PFProducer/plugins/PFProducer.cc

```diff
@@ -333,3 +309,233 @@ void PFProducer::produce(Event& iEvent, const EventSetup& iSetup) {
+    psd_PFMuonAlgo.add<double>("punchThroughMETFactor", 4.0);
+    psd_PFMuonAlgo.add<double>("cosmicRejectionDistance", 1.0);
+    desc.add<edm::ParameterSetDescription>("PFMuonAlgoParameters", psd_PFMuonAlgo);
+  }
+
+  // Input displaced vertices
+  // It is strongly adviced to keep usePFNuclearInteractions = bCorrect
+  desc.add<bool>("rejectTracks_Bad", true);
+  desc.add<bool>("rejectTracks_Step45", true);
+
+  desc.add<bool>("usePFNuclearInteractions", true);
+  desc.add<bool>("usePFConversions", true);
+  desc.add<bool>("usePFDecays", false);
+
+  desc.add<double>("dptRel_DispVtx", 10.0);
+  {
+    edm::ParameterSetDescription psd_CandConnector;
+    psd_CandConnector.add<bool>("bCorrect", true);
+    psd_CandConnector.add<bool>("bCalibPrimary", true);
+    psd_CandConnector.add<double>("dptRel_PrimaryTrack", 10.0);
+    psd_CandConnector.add<double>("dptRel_MergedTrack", 5.0);
+    psd_CandConnector.add<double>("ptErrorSecondary", 1.0);
+    psd_CandConnector.add<std::vector<double>>("nuclCalibFactors", {0.8, 0.15, 0.5, 0.5, 0.05});
+    desc.add<edm::ParameterSetDescription>("iCfgCandConnector", psd_CandConnector);
+  }
+
+  // Treatment of potential fake tracks
+  // Number of sigmas for fake track detection
+  desc.add<double>("nsigma_TRACK", 1.0);
+  // Absolute pt error to detect fake tracks in the first three iterations
+  // dont forget to modify also ptErrorSecondary if you modify this parameter
+  desc.add<double>("pt_Error", 1.0);
+  // Factors to be applied in the four and fifth steps to the pt error
+  desc.add<std::vector<double>>("factors_45", {10.0, 100.0});
+
+  // Treatment of tracks in region of bad HCal
+  desc.add<double>("goodTrackDeadHcal_ptErrRel", 0.2);  // trackRef->ptError()/trackRef->pt() < X
+  desc.add<double>("goodTrackDeadHcal_chi2n", 5);       // trackRef->normalizedChi2() < X
+  desc.add<unsigned int>("goodTrackDeadHcal_layers",
+                         4);                           // trackRef->hitPattern().trackerLayersWithMeasurement() >= X
+  desc.add<double>("goodTrackDeadHcal_validFr", 0.5);  // trackRef->validFraction() > X
+  desc.add<double>("goodTrackDeadHcal_dxy", 0.5);      // [cm] abs(trackRef->dxy(primaryVertex_.position())) < X
+
+  desc.add<double>("goodPixelTrackDeadHcal_minEta", 2.3);    // abs(trackRef->eta()) > X
+  desc.add<double>("goodPixelTrackDeadHcal_maxPt", 50.0);    // trackRef->ptError()/trackRef->pt() < X
+  desc.add<double>("goodPixelTrackDeadHcal_ptErrRel", 1.0);  // trackRef->ptError()/trackRef->pt() < X
+  desc.add<double>("goodPixelTrackDeadHcal_chi2n", 2);       // trackRef->normalizedChi2() < X
+  desc.add<int>(
+      "goodPixelTrackDeadHcal_maxLost3Hit",
+      0);  // max missing outer hits for a track with 3 valid pixel layers (can set to -1 to reject all these tracks)
+  desc.add<int>("goodPixelTrackDeadHcal_maxLost4Hit",
+                1);  // max missing outer hits for a track with >= 4 valid pixel layers
```

Comment:
due to clang-format driven line breaks, it's more practical to put the comment above the line.
Anyways, these seem to fall under comments that better be known to the parameters with ```setComment```